### ISSUE TITLE
`.eslintcache` is deleted when auto-fix is performed

### ISSUE THREAD

**Issue by IanVS** [AtomLinter/linter-eslint#635]
ESLint since version `3.0.0` will no longer remove the `.eslintcache` file whenever a file is linted by linter-eslint.  (see https://github.com/AtomLinter/linter-eslint/issues/587 for discussion of that problem).

_However_, it will delete the cache when we auto-fix a file.  That's because we're not using `--stdin` when we perform fixing (see https://github.com/AtomLinter/linter-eslint/blob/master/src/worker-helpers.js#L109).

I think the solution here is to add an option like `Preserve ESLint cache`, which would add `--cache` when performing a fix.

**Comment by Arcanemagus** [comment]
Is there a reason not to always use that option?

**Comment by IanVS** [comment]
It will also _create_ an `.eslintcache` file, which could be unexpected/undesired.

**Comment by IanVS** [comment]
Actually, it seems that the `.eslintcache` file is cleared on save whether or not `fix-on-save` is turned on.

**Comment by IanVS** [comment]
Note: some changes will need to happen 1-2 levels upstream for this to work correctly.  That's because if we just add the `--cache` flag, the cache will be overwritten with just the one single file that is being saved/fixed, which isn't helpful.

**Comment by IanVS** [comment]
Now that I think about this a little more, instead of an option, how about we look for the presence of an `.eslintcache` file, and if we find one, we add `--cache`?  It wouldn't find custom-named cache files, we could add an option to allow people to specify what filename/pathname they're using.  Not deleting the cache file should just be standard out-of-the-box behavior.

**Comment by Arcanemagus** [comment]
That sounds like a good idea to me, it's too bad we need to wait on at least Atom v1.9.0 stable (if not later versions...) to release built in `eslint@3` 😦.

**Comment by IanVS** [comment]
True, but hopefully most people shouldn't be using the built-in one anyway.  And weren't we talking about getting rid of it at one time?

**Comment by Arcanemagus** [comment]
Yea, I think that's an excellent way to go, unfortunately I haven't had time to really work on this package lately.

**Comment by IanVS** [comment]
Update: the changes were just made to ESLint which will allow us to keep the cache when autofixing.  It will go out with the next ESLint release this Friday.  I'll start working on changes needed to linter-eslint to support it.

**Comment by zimme** [comment]
Would this fix the buffer resetting when performing the auto fix? or do I open a new issue for that?

**Comment by IanVS** [comment]
@zimme, no it would not address that issue, which was already reported in https://github.com/AtomLinter/linter-eslint/issues/620.
